apollo-normalized-cache/src/main/java/com/apollographql/apollo/cache/normalized/ApolloStore.kt

```diff
@@ -0,0 +1,250 @@
+   * @param <D>       type of GraphQL operation data
+   * @param <T>       type operation cached data will be wrapped with
+   * @param <V>       type of operation variables
+   * @return {@ApolloStoreOperation} to be performed, that will be resolved with cached data for specified operation
+  </V></T></D> */
+  fun <D : Operation.Data, T, V : Operation.Variables> read(
+      operation: Operation<D, T, V>): ApolloStoreOperation<T>
+
+  /**
+   * Read GraphQL operation response from store.
+   *
+   * @param operation           response of which should be read
+   * @param responseFieldMapper [ResponseFieldMapper] to be used for field mapping
+   * @param responseNormalizer  [ResponseNormalizer] to be used when reading cached response
+   * @param cacheHeaders        [CacheHeaders] to be used when reading cached response
+   * @param <D>                 type of GraphQL operation data
+   * @param <T>                 type operation cached data will be wrapped with
+   * @param <V>                 type of operation variables
+   * @return {@ApolloStoreOperation} to be performed, that will be resolved with cached response for specified operation
+  </V></T></D> */
+  fun <D : Operation.Data, T, V : Operation.Variables> read(
+      operation: Operation<D, T, V>, responseFieldMapper: ResponseFieldMapper<D>,
+      responseNormalizer: ResponseNormalizer<Record>, cacheHeaders: CacheHeaders): ApolloStoreOperation<Response<T>>
+
+  /**
+   * Read GraphQL fragment from store.
+   *
+   * @param fieldMapper [ResponseFieldMapper] to be used for field mapping
+   * @param cacheKey    [CacheKey] to be used to find cache record for the fragment
+   * @param variables   [Operation.Variables] required for fragment arguments resolving
+   * @param <F>         type of fragment to be read
+   * @return {@ApolloStoreOperation} to be performed, that will be resolved with cached fragment data
+  </F> */
+  fun <F : GraphqlFragment> read(fieldMapper: ResponseFieldMapper<F>,
+                                  cacheKey: CacheKey, variables: Operation.Variables): ApolloStoreOperation<F>
+
+  /**
+   * Write operation to the store.
+   *
+   * @param operation     [Operation] response data of which should be written to the store
+   * @param operationData [Operation.Data] operation response data to be written to the store
+   * @param <D>           type of GraphQL operation data
+   * @param <T>           type operation cached data will be wrapped with
+   * @param <V>           type of operation variables
+   * @return {@ApolloStoreOperation} to be performed, that will be resolved with set of keys of [Record] which
+   * have changed
+  </V></T></D> */
+  fun <D : Operation.Data, T, V : Operation.Variables> write(
+      operation: Operation<D, T, V>, operationData: D): ApolloStoreOperation<Set<String>>
+
+  /**
+   * Write operation to the store and publish changes of [Record] which have changed, that will notify any [ ] that depends on these [Record] to re-fetch.
```

Comment:
This line says `any [ ]`. Is that a mistake? What should `[ ]` have?

Reply:
It should have `ApolloQueryWatcher` but Android Studio isn't able to resolve it. Any idea what might be the issue?